### ISSUE TITLE
Style/SpaceAfterNot breaking for "not"

### ISSUE THREAD

**Issue by NobodysNightmare** [bbatsov/rubocop#3155]
## Expected behavior

`Style/SpaceAfterNot` does not force keyword `not` occurences to lose their space:

* It does **not** report an offense for `not foo`
* It does **not** auto correct it to `notfoo`

## Actual behavior

* It **does** report an offense for `not foo`
* It **does** auto correct it to `notfoo`

## Steps to reproduce the problem

Create a file called `repro.rb`:
````ruby
foo if not bar
````

Run:
````
$ rubocop repro.rb --only Style/SpaceAfterNot
````

(Alternatively disable the cop `Style/Not` via your config file, which should be quite a legit use case)

## RuboCop version

Include the output of `rubocop -V`:

```
$ rubocop -V
0.40.0 (using Parser 2.3.1.0, running on ruby 2.3.0 x86_64-linux)
```

**Comment by NobodysNightmare** [comment]
I am working on a fix...
